### DISCUSSION TITLE
Attaching an event listener on server-side works: Why and How?

### DISCUSSION THREAD

**Discussion by gplusdotgr** [nanojsx/nano#131] [Help / Questions]
```
/** @jsx h */
/// <reference no-default-lib="true"/>
/// <reference lib="dom" />
/// <reference lib="dom.asynciterable" />
/// <reference lib="deno.ns" />

import { serve } from "https://deno.land/std@0.140.0/http/server.ts";
import { h, renderSSR, Helmet } from "https://deno.land/x/nano_jsx@v0.0.20/mod.ts";

function App() {
  return (
    <html>
      <Helmet>
        <html lang="en" amp />
        <title>Nano JSX SSR</title>
        <meta name="description" content="Server Side Rendered Nano JSX Application" />
      </Helmet>
      <body>
        <h1>Hello world</h1>
        <button id="clicker" onClick={() => console.log('1')}>Click me!</button>
      </body>
      <Helmet footer>
        <script>
          document.getElementById('clicker').addEventListener('click', () => alert('clicked'));
        </script>
      </Helmet>
    </html>
  );
}

function handler(req) {
  const html = renderSSR(<App />);
  return new Response(html, {
    headers: {
      "content-type": "text/html",
    },
  });
}

serve(handler);
```

The code below works; :
![image](https://user-images.githubusercontent.com/1768834/214046305-236aac5c-41a3-4980-8a14-d9cb78f513ee.png)

But how? Since I didn't use `hydrate()` .. I can't wrap my head around it. How is a JS event attached to a DOM element on server-side? (there's no DOM to attach to)

**Comment by yandeu** [comment]
There's nothing magic about it. Nanojsx will simply render your jsx to a string and send it to the client.

The event listener will be added on the client-side since there is the following code `<script>document.getElementById('clicker').addEventListener('click', () => alert('clicked'));
</script>`

Btw, I don't think your code renders valid HTML5, because the Helmet tags are not handled properly.
